app/machinedriver_data.go

```diff
@@ -27,7 +27,7 @@ const (
 )
 
 var driverData = map[string]map[string][]string{
-	Amazonec2driver:    {"publicCredentialFields": []string{"accessKey", "region"}, "privateCredentialFields": []string{"secretKey"}},
+	Amazonec2driver:    {"publicCredentialFields": []string{"accessKey", "region"}, "privateCredentialFields": []string{"secretKey", "sessionToken"}},
```

Comment:
@dstockton 

You've added this to cloud credential for ec2 driver. We don't yet support credentials for EKS (or any hosted K8s provider). 

If you just want sessionToken to not be shown in API, you can mark the field as password similar to secret-key here, https://github.com/rancher/kontainer-engine/blob/master/drivers/eks/eks_driver.go#L322